vignettes/Surveys_Walkthrough.Rmd

```diff
@@ -510,78 +715,154 @@ population_data_age <- bind_rows(population_data_age_district_a,
+```{r}
+study_data_cleaned %>% 
+  tabyl(cluster_number, village_name)
 ```
 
+<!-- We can also see from the ordering in the table above, that cluster number is recognized as a character and not as a number, so we fix this with a quick `mutate()` command:   -->
 
-## Browse data
-
-
-## Data cleaning
+<!-- ```{r} -->
+<!-- study_data_cleaned <- study_data_cleaned %>%  -->
+<!--   mutate(cluster_number = as.integer(cluster_number)) -->
+<!-- ``` -->
 
+The fact that there are clusters 1-17 in each of the Towns will complicate later analyses. Each cluster should have a unique identification value. So, let us create one. We use the `str_glue()` function from {stringr} to combine the cluster number and the letter of the town. The letter of the town is extracted from the town name using `str_sub()`, which is also from the {stringr} package. The -1 extracts the last letter of the village name, and places it into the ID.  
 
-### Standardise dates
+```{r}
+study_data_cleaned <- study_data_cleaned %>% 
+  mutate(cluster_number_uid = str_glue(
+    "{town_letter}_{cluster_number}",
+    town_letter = str_sub(village_name, -1)))
+```
 
-The first command creates an list of column names, called `DATEVARS`, using the imported and formatted Kobo data dictionary. The command looks into the `type` column of the dictionary, and extracts all the variables that are dates and that are also present in the `study_data_cleaned`. The names of these columns are saved as `DATEVARS`.  
+We verify:
 
 ```{r}
-## get the names of variables which are dates
-DATEVARS <- study_data_dict %>% 
-  filter(type == "date") %>% 
-  filter(name %in% names(study_data_cleaned)) %>% 
-  ## filter to match the column names of your data
-  pull(name) # select date vars
+study_data_cleaned %>% 
+  tabyl(cluster_number_uid, village_name)
 ```
 
 
-In the next command, from this list `DATEVARS`, and columns that are completely empty are identified. 
+Now, our task is to create a *new* data frame that contains the number of households in each cluster. This is not the number of households chosen to be interviewed, but the universe of *possible* households that could have been chosen in each cluster.  
 
-```{r}
-## find if there are date variables which are completely empty
-EMPTY_DATEVARS <- purrr::map(DATEVARS, ~all(
-  is.na(study_data_cleaned[[.x]])
-  )) %>% 
-  unlist() %>% 
-  which()
-```
+To create a *new* dataframe, the R Markdown template uses the `tibble()` function for your to manually input the names of the clusters, and the number of households. Below, we use a slight adaptation for ease of use: we use the `tribble()` function (note the "r") so that we can type this information in "rows" that look like a normal data frame. Note how the columns are specified at the top with tildes (~).  
 
-Finally, we using indexing (square brackets) and a minus sign to remove those date variables that were empty from the list `DATEVARS`.  
+  
 
+```{r cluster_counts}
```

Comment:
worth offering a shortcut with glue a_ 1:17, b_ 1:17 rep.int 100 length,.... otherwise if ppl have many clusters painful?

Reply:
perhaps but they can just copy/paste for this vignette. Will consider adding when time.